src/App.tsx

```diff
@@ -1,26 +1,43 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { FC, Suspense } from 'react';
+import { Routes, Route } from 'react-router-dom';
+
+import PrivateRoute from './routes/PrivateRoute';
+import Error from './components/pages/Error/Error';
+import Login from './components/pages/Login/Login';
 import './App.css';
 
-function App() {
+const Departments: Function = React.lazy(() => import('./components/pages/Departments/Departments'));
+const Department: Function = React.lazy(() => import('./components/pages/Department/Department'));
+
+const App: FC = () => {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Routes>
+      <Route path="/" element={<Login />} />
+      <Route element={<PrivateRoute />}>
+        <Route
+          path="/departments/:id"
+          element={
+            (
+              <Suspense fallback={<p>Loading...</p>}>
+                <Department />
+              </Suspense>
+            )
+            }
+        />
```

Comment:
could you please check against that example?
https://stackoverflow.com/a/70528778

Reply:
sure, thank you

Reply:
Checked this one, works as expected. I used this syntax to fix an error

**Uncaught Error: [undefined] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>**

It happens when I wrap all my lazy routes in one Suspense component

Reply:
try wrapping with a fragment as well

Reply:
I've tried but got the same error.

Also I've seen the same syntax on the React Router [docs](https://stackblitz.com/github/remix-run/react-router/tree/main/examples/lazy-loading?file=src/App.tsx) so I've decided I can write code like that

Reply:
yeah have seen that too